pymeasure/instruments/temptronic/temptronic_base.py

```diff
@@ -0,0 +1,820 @@
+        t_start = time.time()
+        while(not self.at_temperature()):  # assert at temperature
+            time.sleep(1)
+            t = time.time() - t_start
+
+            tstatus = self.temperature_condition_status_code
+
+            message = ("temp_set= %4.1f deg, "
+                       "temp= %4.1f deg, "
+                       "time= %.2f s, "
+                       "status= %s"
+                       )
+
+            log.info(message,
+                     self.temperature_setpoint,
+                     self.temperature,
+                     t,
+                     tstatus)
+
+            if t > time_limit:
+                log.info('no settling achieved')
+                break
+        log.info('finished this temperature point')
+
+        return self
+
+    def shutdown(self, head=False):
+        """Turn down TS (flow and remote operation).
+
+        :param head: Lift head if ``True``
+
+        :returns: self
+        """
+
+        self.enable_air_flow = 0
+        self.remote_mode = False
+        if head:
+            self.head = 'up'
+        super().shutdown()
+
+        return self
+
+    def start(self, enable_air_flow=True):
+        """start TS in remote mode.
+
+        :param enable_air_flow: flow starts if ``True``
+
+        :returns: self
+        """
+
+        self.remote_mode = 1
+        self.enable_air_flow = enable_air_flow  # enable TS
+
+        return self
+
+    def error_status(self):
+        """Returns error status code (maybe used for logging).
+
+        :returns: :class:`.ErrorCode`
+        """
+        code = self.error_code
+        if not code == 0:
+            log.warning('%s', code)
+        return code
+
+    def cycling_stopped(self):
+        """:returns: ``True`` if cycling has stopped.
+        """
+        return self.temperature_condition_status_code == TemperatureStatusCode.CYCLING_STOPPED
+
+    def end_of_all_cycles(self):
+        """:returns: ``True`` if cycling has stopped.
+        """
+        return self.temperature_condition_status_code == TemperatureStatusCode.END_OF_ALL_CYCLES
+
+    def end_of_one_cycle(self):
+        """:returns: ``True`` if TS is at end of one cycle.
+        """
+        return self.temperature_condition_status_code == TemperatureStatusCode.END_OF_ONE_CYCLE
+
+    def end_of_test(self):
+        """:returns: ``True`` if TS is at end of test.
+        """
+        return self.temperature_condition_status_code == TemperatureStatusCode.END_OF_TEST
+
+    def not_at_temperature(self):
+        """:returns: ``True`` if not at temperature.
+        """
+        return self.temperature_condition_status_code == TemperatureStatusCode.NOT_AT_TEMPERATURE
+
+    def at_temperature(self):
+        """:returns: ``True`` if at temperature.
+        """
+        return self.temperature_condition_status_code == TemperatureStatusCode.AT_TEMPERATURE
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self):
+        self.shutdown()
```

Comment:
These methods have [landed in master](https://github.com/pymeasure/pymeasure/blob/master/pymeasure/instruments/instrument.py#L165-L169) by now, so should be removed.